parser-typechecker/src/Unison/Builtin.hs

```diff
@@ -156,140 +160,182 @@ builtinTypeDependents r = Rel.lookupRan r builtinDependencies
-      , ("Bytes.toSequence", "Bytes -> [Nat]")
-      , ("Bytes.size", "Bytes -> Nat")
-      , ("Bytes.flatten", "Bytes -> Bytes")
-
-      , ("Sequence.empty", "[a]")
-      , ("Sequence.cons", "a -> [a] -> [a]")
-      , ("Sequence.snoc", "[a] -> a -> [a]")
-      , ("Sequence.take", "Nat -> [a] -> [a]")
-      , ("Sequence.drop", "Nat -> [a] -> [a]")
-      , ("Sequence.++", "[a] -> [a] -> [a]")
-      , ("Sequence.size", "[a] -> Nat")
-      , ("Sequence.at", "Nat -> [a] -> Optional a")
-
-      , ("Debug.watch", "Text -> a -> a")
-      , ("Effect.pure", "a -> Effect e a") -- Effect ambient e a
-      , ("Effect.bind", "'{e} a -> (a ->{ambient} b) -> Effect e a") -- Effect ambient e a
+      , ("Universal.compare", forall1 "a" (\a -> a --> a --> int))
+      , ("Universal.>", forall1 "a" (\a -> a --> a --> boolean))
+      , ("Universal.<", forall1 "a" (\a -> a --> a --> boolean))
+      , ("Universal.>=", forall1 "a" (\a -> a --> a --> boolean))
+      , ("Universal.<=", forall1 "a" (\a -> a --> a --> boolean))
+
+      , ("Boolean.not", boolean --> boolean)
+
+      , ("Text.empty", text)
+      , ("Text.++", text --> text --> text)
+      , ("Text.take", nat --> text --> text)
+      , ("Text.drop", nat --> text --> text)
+      , ("Text.size", text --> nat)
+      , ("Text.==", text --> text --> boolean)
+      , ("Text.!=", text --> text --> boolean)
+      , ("Text.<=", text --> text --> boolean)
+      , ("Text.>=", text --> text --> boolean)
+      , ("Text.<", text --> text --> boolean)
+      , ("Text.>", text --> text --> boolean)
+
+      , ("Bytes.empty", bytes)
+      , ("Bytes.fromSequence", sequence nat --> bytes)
+      , ("Bytes.++", bytes --> bytes --> bytes)
+      , ("Bytes.take", nat --> bytes --> bytes)
+      , ("Bytes.drop", nat --> bytes --> bytes)
+      , ("Bytes.at", nat --> bytes --> optional nat)
+      , ("Bytes.toSequence", bytes --> sequence nat)
+      , ("Bytes.size", bytes --> nat)
+      , ("Bytes.flatten", bytes --> bytes)
+
+      , ("Sequence.empty", forall1 "a" (\a -> sequence a))
+      , ("Sequence.cons", forall1 "a" (\a -> a --> sequence a --> sequence a))
+      , ("Sequence.snoc", forall1 "a" (\a -> sequence a --> a --> sequence a))
+      , ("Sequence.take", forall1 "a" (\a -> nat --> sequence a --> sequence a))
+      , ("Sequence.drop", forall1 "a" (\a -> nat --> sequence a --> sequence a))
+      , ("Sequence.++", forall1 "a" (\a -> sequence a --> sequence a --> sequence a))
+      , ("Sequence.size", forall1 "a" (\a -> sequence a --> nat))
+      , ("Sequence.at", forall1 "a" (\a -> nat --> sequence a --> optional a))
+
+      , ("Debug.watch", forall1 "a" (\a -> text --> a --> a))
+      , ("Effect.pure", forall2 "a" "e" (\a e -> a --> effect e a)) -- Effect ambient e a
+      , ("Effect.bind", forall4 "e" "a" "ambient" "b" (\e a ambient b -> delayed (effectful e a) --> (a --> effectful ambient b) --> effect e a)) -- Effect ambient e a
```

Comment:
Even though the tests pass, I'm not sure I got the type of `Effect.bind` right. What is the full version (with explicit `forall`s) of the original `'{e} a -> (a ->{ambient} b) -> Effect e a`?